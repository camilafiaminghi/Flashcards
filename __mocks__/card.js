const card = {
  question: 'What is React?',
  answer: 'A library for managing user interfaces'
}

export default card
