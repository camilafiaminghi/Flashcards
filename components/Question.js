import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { textColorInverse, textColor, pColor, pColorLight, pColorDark, sColor, sColorLight, sColorDark } from '../utils/colors'

const Question = ({ question, toggleSide }) => {
	return (
		<View style={styles.container}>
			<Text style={styles.text}>{question}</Text>

			<TouchableOpacity
				style={[styles.input, styles.btn]}
				onPress={toggleSide}>
				<Text style={styles.btnText}>{'Show Answer'.toUpperCase()}</Text>
			</TouchableOpacity>
		</View>
	)
}

Question.propTypes = {
	question: PropTypes.string.isRequired,
	toggleSide: PropTypes.func.isRequired
}

export default Question

/* STYLES */
export const styles = StyleSheet.create({
  container: {
    flex: 1,
  	flexGrow: 1,
  	alignItems: 'center',
  	justifyContent: 'center'
  },
  text: {
  	textAlign: 'center',
		fontSize: 28,
		marginRight: 10,
		marginLeft: 10,
		marginBottom: 20,
		color: textColor
  },
  input: {
		flexDirection: 'row',
		alignSelf: 'stretch',
		height: 50,
	},
	btn: {
		alignItems: 'center',
		justifyContent: 'center',
		padding: 4,
		marginRight: 10,
		marginLeft: 10
	},
	btnText: {
		fontSize: 14,
		fontWeight: 'bold',
		color: pColorDark
	}
})
