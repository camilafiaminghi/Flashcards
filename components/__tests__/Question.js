import React, { Component } from 'react'
import { TouchableOpacity } from 'react-native'
import renderer from 'react-test-renderer'
import Question from '../Question'
import card from '../../__mocks__/card'

let wrapper
let instance
let component
const props = {
	question: card.question,
	toggleSide: jest.fn()
}

describe('<Question />', () => {

	beforeEach(() => {
		component = shallow(<Question {...props} />)
	})

	it('matches snapshot', () => {
    wrapper = renderer.create(<Question {...props} />)
    expect(wrapper.toJSON()).toMatchSnapshot()
  });

	it('should render', () => {
		expect(wrapper).toBeTruthy()
	})

	it('should call toggleSide onPress TouchableOpacity', () => {
		const button = component.find(TouchableOpacity).get(0)
		button.props.onPress()

		expect(props.toggleSide).toHaveBeenCalledTimes(1)
	})
})
