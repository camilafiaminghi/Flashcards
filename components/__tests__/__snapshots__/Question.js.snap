// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333333",
        "fontSize": 28,
        "marginBottom": 20,
        "marginLeft": 10,
        "marginRight": 10,
        "textAlign": "center",
      }
    }
  >
    What is React?
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "marginLeft": 10,
        "marginRight": 10,
        "opacity": 1,
        "padding": 4,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#6f79a8",
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      SHOW ANSWER
    </Text>
  </View>
</View>
`;
